docs/source/func.whirlwind_tour.rst

```diff
@@ -0,0 +1,197 @@
+runs on examples and then lift it to a function that can take batches of
+examples with ``vmap(func)``, leading to a simpler modeling experience:
+
+.. code-block:: python
+
+    import torch
+    from torch.func import vmap
+    batch_size, feature_size = 3, 5
+    weights = torch.randn(feature_size, requires_grad=True)
+
+    def model(feature_vec):
+        # Very simple linear model with activation
+        assert feature_vec.dim() == 1
+        return feature_vec.dot(weights).relu()
+
+    examples = torch.randn(batch_size, feature_size)
+    result = vmap(model)(examples)
+
+When composed with :func:`grad`, :func:`vmap` can be used to compute per-sample-gradients:
+
+.. code-block:: python
+
+    from torch.func import vmap
+    batch_size, feature_size = 3, 5
+
+    def model(weights,feature_vec):
+        # Very simple linear model with activation
+        assert feature_vec.dim() == 1
+        return feature_vec.dot(weights).relu()
+
+    def compute_loss(weights, example, target):
+        y = model(weights, example)
+        return ((y - target) ** 2).mean()  # MSELoss
+
+    weights = torch.randn(feature_size, requires_grad=True)
+    examples = torch.randn(batch_size, feature_size)
+    targets = torch.randn(batch_size)
+    inputs = (weights,examples, targets)
+    grad_weight_per_example = vmap(grad(compute_loss), in_dims=(None, 0, 0))(*inputs)
+
+:func:`vjp` (vector-Jacobian product)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :func:`vjp` transform applies ``func`` to ``inputs`` and returns a new function
+that computes the vector-Jacobian product (vjp) given some ``cotangents`` Tensors.
+
+.. code-block:: python
+
+    from torch.func import vjp
+
+    inputs = torch.randn(3)
+    func = torch.sin
+    cotangents = (torch.randn(3),)
+
+    outputs, vjp_fn = vjp(func, inputs); vjps = vjp_fn(*cotangents)
+
+:func:`jvp` (Jacobian-vector product)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :func:`jvp` transforms computes Jacobian-vector-products and is also known as
+"forward-mode AD". It is not a higher-order function unlike most other transforms,
+but it returns the outputs of ``func(inputs)`` as well as the jvps.
+
+.. code-block:: python
+
+    from torch.func import jvp
+    x = torch.randn(5)
+    y = torch.randn(5)
+    f = lambda x, y: (x * y)
+    _, output = jvp(f, (x, y), (torch.ones(5), torch.ones(5)))
+    assert torch.allclose(output, x + y)
+
+:func:`jacrev`, :func:`jacfwd`, and :func:`hessian`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :func:`jacrev` transform returns a new function that takes in ``x`` and returns
+the Jacobian of the function with respect to ``x`` using reverse-mode AD.
+
+.. code-block:: python
+
+    from torch.func import jacrev
+    x = torch.randn(5)
+    jacobian = jacrev(torch.sin)(x)
+    expected = torch.diag(torch.cos(x))
+    assert torch.allclose(jacobian, expected)
+
+Use :func:`jacrev` to compute the jacobian. This can be composed with vmap to produce
```

Comment:
Nit: the wording is a little weird here. Maybe it meant to be either: "`jacrev` can be composed with ..."